lib/io/hardware_i2c.cpp

```diff
@@ -69,11 +69,31 @@ std::optional<std::uint8_t> HardwareI2c::readByte(const std::uint8_t device_addr
 core::Result HardwareI2c::writeByteToRegister(const std::uint8_t device_address,
                                               const std::uint8_t register_address,
                                               const std::uint8_t data)
+{
+  const std::uint8_t register_address_array[2] = {0x00, register_address};
+  return writeByteToDevice(device_address, register_address_array, data);
+}
+
+core::Result HardwareI2c::writeByteToRegister(const std::uint8_t device_address,
+                                              const std::uint16_t register_address,
+                                              const std::uint8_t data)
+{
+  // TODOLater - Is this the best way to do this?
```

Comment:
this is the best way to do it i'm pretty sure

Reply:
Seemed slightly hacky, but fair enough